cpg_workflows/jobs/trim.py

```diff
@@ -0,0 +1,239 @@
+"""
+Trim raw FASTQ reads using cutadapt
+"""
+
+import hailtop.batch as hb
+from hailtop.batch.job import Job
+from cpg_utils.hail_batch import command, image_path
+from cpg_utils.config import get_config
+from cpg_workflows.utils import can_reuse
+from cpg_workflows.resources import STANDARD
+from cpg_workflows.filetypes import (
+    FastqPair,
+    FastqPairs,
+)
+from cpg_workflows.workflow import (
+    SequencingGroup,
+)
+from dataclasses import dataclass
+from enum import Enum
+
+
+class MissingFastqInputException(Exception):
+    """Raise if alignment input is missing"""
+    pass
+
+
+class InvalidSequencingTypeException(Exception):
+    """Raise if alignment type is not 'rna'"""
+    pass
+
+
+@dataclass
+class AdapterSequence:
+    """
+    A class to represent an adapter sequence.
+    """
+
+    sequence: str
+    name: str | None = None
+
+
+@dataclass
+class AdapterPair:
+    """
+    A class to represent a pair of adapter sequences.
+    """
+
+    r1: AdapterSequence
+    r2: AdapterSequence
+
+
+class AdapterPairs(Enum):
+    """
+    A class to represent a set of adapter pairs.
+    """
+
+    ILLUMINA_TRUSEQ = AdapterPair(
+        r1=AdapterSequence(
+            sequence='AGATCGGAAGAGCACACGTCTGAACTCCAGTCA',
+            name='TruSeq Adapter Index 1',
+        ),
+        r2=AdapterSequence(
+            sequence='AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT',
+            name='TruSeq Adapter Index 2',
+        ),
+    )
+
+
+class Cutadapt:
+    """
+    Construct a cutadapt command for trimming FASTQs.
+    """
+
+    def __init__(
+        self,
+        input_fastq_pair: FastqPair,
+        output_fastq_pair: FastqPair,
+        adapter_type: str,
+        paired: bool = True,
+        min_length: int = 50,
+        two_colour: bool = True,
+        polyA: bool = False,
+        quality_trim: int | None = None,  # TODO: determine default
+    ):
+        try:
+            adapters: AdapterPair = AdapterPairs[adapter_type].value
+        except AttributeError:
+            raise ValueError(f'Invalid adapter type: {adapter_type}')
+        self.command = [
+            'cutadapt',
```

Comment:
It is a matter of style, but using f-strings to build cmd blicks is generally preferred.